presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1024,14 +1028,14 @@ private boolean shouldRepartitionForIndexJoin(List<VariableReferenceExpression>
 
             // Disable repartitioning if it would disrupt a parent's partitioning preference when streaming is enabled
             boolean parentAlreadyPartitionedOnChild = parentPartitioningPreferences
-                    .map(partitioning -> probeProperties.isStreamPartitionedOn(partitioning.getPartitioningColumns()))
+                    .map(partitioning -> probeProperties.isStreamPartitionedOn(partitioning.getPartitioningColumns(), isExactPartitioningPreferred(session)))
```

Comment:
It looks like we are specifically trying to avoid an extra partitioning. Given the IndexJoin doesn't use any extra memory, should it be left as is?